lib/eclipse/EclipseState/Schedule/TimeMap.cpp

```diff
@@ -88,36 +93,56 @@ namespace Opm {
     {
         if (m_timeList.size() < 2)
             return 0.0;
-        boost::posix_time::time_duration deltaT = m_timeList.back() - m_timeList.front();
-        return static_cast<double>(deltaT.total_milliseconds())/1000.0;
+        const time_t deltaT = m_timeList.back() - m_timeList.front();
+        return static_cast<double>(deltaT);
     }
 
+    void TimeMap::addTime(std::time_t newTime) {
+        const std::time_t lastTime = m_timeList.back();
+        const size_t step = m_timeList.size();
+        if (newTime > lastTime) {
+            const std::tm new_date = *gmtime(&newTime);
+            const std::tm prev_date = *gmtime(&lastTime);
+
+            if (new_date.tm_mon != prev_date.tm_mon)
+                m_first_timestep_months.push_back(step);
+
+            if (new_date.tm_year != prev_date.tm_year)
+                m_first_timestep_years.push_back(step);
+
+            m_timeList.push_back(newTime);
+        } else
+            throw std::invalid_argument("Times added must be in strictly increasing order.");
+    }
 
     void TimeMap::addTime(boost::posix_time::ptime newTime) {
-        boost::posix_time::ptime lastTime = m_timeList.back();
+        boost::posix_time::ptime lastTime = boost::posix_time::from_time_t(m_timeList.back());
         size_t step = m_timeList.size();
         if (newTime > lastTime) {
             boost::gregorian::date new_date = newTime.date();
             boost::gregorian::date prev_date = lastTime.date();
 
             if (new_date.month() != prev_date.month())
-              m_first_timestep_months.push_back(step);
+                m_first_timestep_months.push_back(step);
 
             if (new_date.year() != prev_date.year())
                 m_first_timestep_years.push_back( step );
 
-            m_timeList.push_back( newTime );
+            m_timeList.push_back(to_time_t(newTime));
         } else
             throw std::invalid_argument("Times added must be in strictly increasing order.");
     }
 
+    void TimeMap::addTStep(std::time_t step) {
```

Comment:
If you're looking for 64 bit support, `long` is not sufficient. It's only guaranteed to be at least 32 bit.